File: design-patterns/go/adapter/file_api/third_party_api.go
// third_party_api.go

package main

import "fmt"

type ThirdPartyApi struct {
}

func NewThirdPartyApi() (thirdPartyApi *ThirdPartyApi) {
	thirdPartyApi = &ThirdPartyApi{}
	return
}

func (thirdPartyApi *ThirdPartyApi) FetchData() (string) {
	// code to fetch data from Third Party API

	fmt.Println("Fetching data from Third Party API")
	return "Data read from Third Party Api"
}

func (thirdPartyApi *ThirdPartyApi) SendData(data string) {
	// code to send data to Third Party API

	fmt.Printf("Sending data to Third Party API: %v\n", data)
}

File: design-patterns/php/adapter/fileadapter/ThirdPartyApi.php
<?php
// ThirdPartyApi.php

namespace BigBoxCode\DesignPattern\Adapter\FileAdapter;


class ThirdPartyApi implements Api {
    public function fetchData(): string {
        echo "Fetching data from Third Party API\n";

        return "Data read from Third Party Api";
    }

    public function sendData(string $data): void {
        echo "Sending data to Third Party API: " . $data . "\n";
    }
}